Tests/js/models/assets/ez-locationmodel-tests.js

```diff
@@ -934,6 +934,45 @@ YUI.add('ez-locationmodel-tests', function (Y) {
         },
     });
 
+    swapTest = new Y.Test.Case({
+        "name": "eZ Location Swap tests",
+
+        setUp: function () {
+            this.capiMock = new Y.Mock();
+            this.capiGetService = 'getContentService';
+            this.locationId = 'location/man-city/sergio-aguero';
+            this.destinationLocationId = 'location/fcb/arda-turam';
+            this.model = new Y.eZ.Location({id: this.locationId});
+            this.destinationLocation = new Y.eZ.Location({id: this.destinationLocationId});
+            this.contentServiceMock = new Y.Mock();
+        },
+
+        tearDown: function () {
+            this.model.destroy();
+            delete this.model;
+        },
+
+        "Should swap the location": function () {
+            var callback = function () {},
+                options = {api: this.capiMock};
+
+            Y.Mock.expect(this.capiMock, {
+                method: 'getContentService',
+                returns: this.contentServiceMock
+            });
+            Y.Mock.expect(this.contentServiceMock, {
+                method: 'swapLocation',
+                args: [this.locationId, this.destinationLocationId, Y.Mock.Value.Function],
```

Comment:
with https://github.com/ezsystems/PlatformUIBundle/pull/592/files#r63542286 taken into account, you should replace `Y.Mock.Value.Function` by `callback` to make sure the callback is passed to the JS REST Client method.